### PR TITLE
Adding with-ably example for realtime messaging in next apps

### PR DESCRIPTION
**Pull request by thisisjofrank**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

**Comment by leerob** [comment]
Could you include a deployed example here so we can test it? 😄

**Comment by thisisjofrank** [comment]
Deployed to vercel here: https://next-and-ably.vercel.app/

**Review by leerob — COMMENTED** [review]
LGTM!

File: examples/with-ably/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Realtime Edge Messaging with [Ably](https://ably.com/)
+
```
[the remark this answers is not in the document]
Comment (thisisjofrank) [reply examples/with-ably/README.md:2]: :) done

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, could you convert this to TypeScript as per our examples contribution guide? https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

**Comment by thisisjofrank** [comment]
Updated to use typescript

File: examples/with-ably/package.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
+  "name": "next-and-ably",
+  "version": "0.1.0",
```
Comment (leerob) [inline examples/with-ably/package.json:3]:
```suggestion
```

File: examples/with-ably/package.json [hunk 3]
```diff
@@ -8,13 +11,16 @@
   "dependencies": {
     "@ably-labs/react-hooks": "^2.0.1",
     "ably": "^1.2.22",
-    "next": "latest",
+    "got": "^12.1.0",
+    "next": "12.1.6",
```
Comment (leerob) [inline examples/with-ably/package.json:15]:
```suggestion
    "next": "latest",
```

File: examples/with-ably/package.json [hunk 4]
```diff
@@ -8,13 +11,16 @@
...
     "react": "18.1.0",
     "react-dom": "18.1.0"
   },
   "devDependencies": {
-    "eslint": "latest",
-    "eslint-config-next": "latest",
-    "next-transpile-modules": "^9.0.0"
+    "@types/react": "^18.0.12",
+    "eslint": "8.16.0",
+    "eslint-config-next": "12.1.6",
```
Comment (leerob) [inline examples/with-ably/package.json:22]:
```suggestion
    "eslint-config-next": "latest",
```

**Comment by thisisjofrank** [comment]
> Thanks, could you convert this to TypeScript as per our examples contribution guide? https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

This is now done and up to date.

File: examples/with-ably/package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    "next": "latest",
+    "react": "18.1.0",
+    "react-dom": "18.1.0"
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.12",
+    "eslint": "8.16.0",
+    "eslint-config-next": "latest",
+    "next-transpile-modules": "^9.0.0",
+    "typescript": "^4.7.3"
```
Comment (balazsorban44) [inline examples/with-ably/package.json:22]:
```suggestion
    "start": "next start"
  },
  "dependencies": {
    "@ably-labs/react-hooks": "^2.0.1",
    "ably": "^1.2.22",
    "got": "^12.1.0",
    "next": "latest",
    "react": "18.1.0",
    "react-dom": "18.1.0"
  },
  "devDependencies": {
    "@types/react": "^18.0.12",
    "next-transpile-modules": "^9.0.0",
    "typescript": "^4.7.3"
```

`@types/node` is also missing.

File: examples/with-ably/pages/types.d.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export type TextMessage = { text: string };
```
Comment (balazsorban44) [inline examples/with-ably/pages/types.d.ts:1]: This file should probably be outside `pages`

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Added some more comments, please have a look

File: examples/with-ably/next.config.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true
+};
+
+const withTM = require("next-transpile-modules")(["@ably-labs/react-hooks"]);
+
+module.exports = withTM(nextConfig);
```
[the remark this answers is not in the document]
Comment (thisisjofrank) [reply examples/with-ably/next.config.js]: removed, ably-react-hooks package updated

File: examples/with-ably/pages/api/createTokenRequest.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import Ably from "ably/promises";
+import { NextApiRequest, NextApiResponse } from "next";
```
[the remark this answers is not in the document]
Comment (thisisjofrank) [reply examples/with-ably/pages/api/createTokenRequest.ts]: updated

File: examples/with-ably/pages/types.d.ts (hunk 6 above)
Comment (thisisjofrank) [reply examples/with-ably/pages/types.d.ts]: moved

**Review by leerob — DISMISSED** [review]

**Review by balazsorban44 — DISMISSED** [review]
Thanks!

**Review by ijjk — COMMENTED** [review]
Hi, please run `pnpm lint-fix` to resolve the failing checks! Seems we aren't able to update this branch as well since changes from contributors is disabled.

**Comment by thisisjofrank** [comment]
> Hi, please run `pnpm lint-fix` to resolve the failing checks! Seems we aren't able to update this branch as well since changes from contributors is disabled.

I've run lint-fix, it gave me over 7000 changes, I've committed only the ones to the files I've added in this branch.

**Review by leerob — APPROVED** [review]

**Pull request merged**